lib/cisco_node_utils/ace.rb

```diff
@@ -0,0 +1,225 @@
+            item.proto = match[3]
+            item.src_addr = match[4]
+            item.src_port = match[5]
+            item.dst_addr = match[6]
+            item.dst_port = match[7]
+            item.option_format = match[8]
+            hash[name][match[1].to_i] = item
+          end
+        end
+      end
+      hash
+    end
+
+    # Destroy a router instance; disable feature on last instance
+    def destroy
+      config_ace('no')
+    end
+
+    # First create acl
+    def config_ace(state='')
+      @set_args[:state] = state
+      config_set('acl', 'ace', @set_args)
+    end
+
+    # ----------
+    # PROPERTIES
+    # ----------
+    # getter of seqno
+    def seqno
+      str = config_get('acl', 'ace', @get_args)
+      return if str.nil?
+
+      regexp = Regexp.new(@regexp_str)
+      match = regexp.match(str)
+      match[1]
+    end
+
+    # setter of seqno
+    def seqno=(state)
+      @set_args[:state] = state ? '' : 'no'
+    end
+
+    # getter of action
+    def action
+      str = config_get('acl', 'ace', @get_args)
+      return if str.nil?
+
+      regexp = Regexp.new(@regexp_str)
+      match = regexp.match(str)
+      match[2]
+    end
+
+    # setter of action
+    def action=(action)
+      @set_args[:action] = action
+    end
+
+    # getter of proto
+    def proto
+      str = config_get('acl', 'ace', @get_args)
+      return if str.nil?
+
+      regexp = Regexp.new(@regexp_str)
+      match = regexp.match(str)
+      match[3]
+    end
+
+    # setter of proto
+    def proto=(proto)
+      @set_args[:proto] = proto
+    end
+
+    # getter of src_addr
+    def src_addr
+      str = config_get('acl', 'ace', @get_args)
+      return if str.nil?
+
+      regexp = Regexp.new(@regexp_str)
+      match = regexp.match(str)
+      match[4]
+    end
+
+    # setter of src_addr
+    def src_addr=(src_addr)
+      @set_args[:src_addr] = src_addr
+    end
+
+    # getter of src_port
+    def src_port
+      str = config_get('acl', 'ace', @get_args)
+      return if str.nil?
+
+      regexp = Regexp.new(@regexp_str)
+      match = regexp.match(str)
+      match[5]
+    end
+
+    # setter of src_port_format
+    def src_port=(src_port)
+      @set_args[:src_port] = src_port
+    end
+
+    # getter of dst_addr_format
+    def dst_addr
+      str = config_get('acl', 'ace', @get_args)
+      return if str.nil?
+
+      regexp = Regexp.new(@regexp_str)
+      match = regexp.match(str)
+      match[6]
+    end
+
+    # setter of dst_addr_format
+    def dst_addr=(dst_addr)
+      @set_args[:dst_addr] = dst_addr
+    end
+
+    # getter of dst_port_format
+    def dst_port
+      str = config_get('acl', 'ace', @get_args)
+      return if str.nil?
+
+      regexp = Regexp.new(@regexp_str)
+      match = regexp.match(str)
+      match[7]
+    end
+
+    # setter of dst_port_format
+    def dst_port=(src_port)
+      @set_args[:dst_port] = src_port
+    end
+
+    # getter of option_format
+    def option_format
```

Comment:
Let's discuss this one after the rewrites above. It will have to change but I'm mulling over some options.